### PR TITLE
Update to conda build 3 and cmake to 3.12.0

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
This PR is just to do some tests in windows

File: recipe/bld.bat [hunk 1]
```diff
@@ -11,12 +11,18 @@ set CMAKE_CONFIG="Release"
...
     -DCMAKE_BUILD_TYPE=%CMAKE_CONFIG%                        ^
     -DCMAKE_FIND_ROOT_PATH="%LIBRARY_PREFIX%"                ^
+    -DCMAKE_PREFIX_PATH="%PREFIX%"                           ^
+    -DCMAKE_CXX_STANDARD:STRING=17                                  ^
+    -DCMake_HAVE_CXX_MAKE_UNIQUE:INTERNAL=TRUE ^
+    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON                         ^
     -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ..
 if errorlevel 1 exit 1
 
+cmake --build . --config %CMAKE_CONFIG% --target ALL_BUILD
```
Comment (isuruf) [inline recipe/bld.bat]: Is this necessary?

Comment (marcelotrevisani) [reply recipe/bld.bat]: Not anymore

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,40 +1,45 @@
...
+{% set version = "3.12.0" %}
 {% set major_minor = '.'.join(version.split(".")[:2]) %}
 
 package:
   name: cmake
   version: {{ version }}
 
 source:
-  url: https://cmake.org/files/v{{ major_minor }}/cmake-{{ version }}.tar.gz
+  url: https://gitlab.kitware.com/cmake/cmake/-/archive/v{{ version }}/cmake-v3.12.0.tar.bz2
```
Comment (scopatz) [inline recipe/meta.yaml]: The last bit here should be `cmake-v{{ version }}.tar.bz2`

Comment (marcelotrevisani) [reply recipe/meta.yaml]: Yes sure!

**Comment by scopatz** [comment]
Ok this looks like it passed! But why did it only run on appveyor?

**Comment by marcelotrevisani** [comment]
Hi @scopatz I created this PR just to test in windows because it was failing just in this platform. My idea here was to find what were necessary to add into the recipe or the build script to complement the another PR which the conda's maintainer are doing.
I was just trying to help. You can close this if you want to.  :)

**Comment by scopatz** [comment]
Oh no worries, just please open a full PR, please.

**Comment by marcelotrevisani** [comment]
@scopatz , please review
:)

**Comment by scopatz** [comment]
Thanks @marcelotrevisani!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
-  url: https://cmake.org/files/v{{ major_minor }}/cmake-{{ version }}.tar.gz
+  url: https://gitlab.kitware.com/cmake/cmake/-/archive/v{{ version }}/cmake-v{{ version }}.tar.bz2
   fn: cmake-{{ version }}.tar.gz
-  sha256: 8f864e9f78917de3e1483e256270daabc4a321741592c5b36af028e72bff87f5
+  sha256: 461dd6b49d636a044f65efcec0be2f448e531d3ac593966944407c3198bbb334
 
 build:
   number: 0
-  msvc_compiler: 14.0
+  skip: True  # [win and vc<14]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we be just skipping `vc!=14`?

Comment (scopatz) [reply recipe/meta.yaml]: It was failing to build at all with vc<14.  This at least gets us moving again and able to migrate to the new compilers

**Pull request merged**